### PR TITLE
Cache problem fix

### PR DESCRIPTION
**Pull request by ilya-shaydullin**
- Fixed cache problem with the same file name
- Add "or", "and" and "raw" helpers
- Added tests

### CODE REVIEW

File: shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/PathLocator.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package dev.icerock.tools.shaper.core
```
Comment (Alex009) [inline shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/PathLocator.kt:5]: copyright required

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/PathLocator.kt (hunk 1 above)
Comment (ilya-shaydullin) [reply shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/PathLocator.kt:5]: Fixed

**Review by Alex009 — APPROVED** [review]

**Comment by Alex009** [comment]
@ilusha2012 need to resolve conflicts

File: shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/YamlConfigReader.kt [hunk 2]
```diff
@@ -63,13 +63,14 @@ object YamlConfigReader {
         val filesList = files as? List<Map<String, Any>>
         val filesDirectory = files as? String
         return filesList?.map { fileMap ->
-            val templateFile = File(directory, fileMap["contentTemplateName"] as String)
+            PathLocator.set(fileMap["contentTemplateName"] as String, directory.absolutePath)
             TemplateConfig.FileConfig(
                 pathTemplate = fileMap["pathTemplate"] as String,
-                contentTemplateName = templateFile.path,
+                contentTemplateName = fileMap["contentTemplateName"] as String,
```
Comment (Alex009) [inline shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/YamlConfigReader.kt]: save this path to variable, not duplicate getByKey access please

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/YamlConfigReader.kt (hunk 2 above)
Comment (ilya-shaydullin) [reply shaper-core/src/main/kotlin/dev/icerock/tools/shaper/core/YamlConfigReader.kt]: Done

**Review by Alex009 — APPROVED** [review]

**Pull request merged**